sycl/source/detail/program_impl.cpp

```diff
@@ -238,23 +238,6 @@ void program_impl::compile_with_kernel_name(std::string KernelName,
   MState = program_state::compiled;
 }
 
-void program_impl::build_with_kernel_name(std::string KernelName,
-                                          std::string BuildOptions) {
-  std::lock_guard<std::mutex> Lock(MMutex);
-  throw_if_state_is_not(program_state::none);
-  if (!is_host()) {
-    MProgramAndKernelCachingAllowed = true;
-    MBuildOptions = BuildOptions;
-    MProgram = ProgramManager::getInstance().getBuiltPIProgram(
-        detail::getSyclObjImpl(get_context()),
-        detail::getSyclObjImpl(get_devices()[0]), KernelName, this,
-        /*JITCompilationIsRequired=*/(!BuildOptions.empty()));
-    const PluginPtr &Plugin = getPlugin();
-    Plugin->call<PiApiKind::piProgramRetain>(MProgram);
-  }
-  MState = program_state::linked;
-}
-
```

Comment:
I think it makes sense to remove `is_cacheable` as well, since it's always false now.

Reply:
Done.